src/cmdlib.sh

```diff
@@ -275,6 +273,19 @@ prepare_build() {
         fi
     done
 
+    # For Fedora CoreOS pick up the stream from the manifest.yaml in the top SCM
+    # and pass that in to `write_image_json` to template `{stream}` inside
+    # image.yaml fields.
+    # context: https://github.com/coreos/fedora-coreos-releng-automation/pull/202#issuecomment-2770596788
+    stream=""
+    if [ "${name}" == "fedora-coreos" ]; then
+        yaml2json "${manifest}" "${tmp_builddir}/manifest-get-stream.json"
+        stream=$(jq -r '.variables.stream' < "${tmp_builddir}/manifest-get-stream.json")
+        rm "${tmp_builddir}/manifest-get-stream.json"
+    fi
+    export image_json="${tmp_builddir}/image.json"
```

Comment:
I'm OK with this, though was thinking of something a little more generic where every variable in `manifest.yaml` is just automatically picked up. That also avoids hardcoding `fedora-coreos`, which I prefer when we can.

I guess one easy way to do that is to instead of special-casing the stream and passing that down to `write_image_json`, we pass down the path to `manifest.yaml` instead, and then in `generate_image_json`, since we're in Python-land there, we open `manifest.yaml`, get the `variables` dict, and pass that down to `flatten_image_yaml`, which can then use it to format the contents.

Reply:
ok.. implemented.